main.go

```diff
@@ -106,15 +108,17 @@ func udpServer(address string, h *hub) {
 	}
 }
 
-func sender(c *net.UDPConn, s <-chan sendPacket) {
+func sender(c *net.UDPConn, s <-chan SendPacket) {
 	rateLogger := loggers.NewRateLogger(100)
 	for message := range s {
 		_, err := c.WriteToUDP(message.data, message.addr)
 
 		rateLogger.Log("Sender frequency", log.Fields{})
 
 		if err != nil {
-			log.Printf("Error: UDP write: %v", err)
+            log.WithFields(log.Fields{
+                "error": err,
+            }).Fatal("UDP write error")
 		}
 	}
 }
```

Comment:
Again can't comment on the actual line, but use logrus for logging inside main, specifically Lines 136, 151, and 165